test/lib/modules/endpointRunner.test.js

```diff
@@ -0,0 +1,367 @@
+        return done();
+      });
+    });
+
+    it('should include empty object as body if null body passed', (done) => {
+      nock('https://baas.kinvey.com')
+        .matchHeader('content-type', 'application/json')
+        .matchHeader('x-kinvey-api-version', '3')
+        .post(`/rpc/${environmentId}/custom/myEndpoint`, {})
+        .basicAuth({
+          user: environmentId,
+          pass: mastersecret
+        })
+        .reply(200, [{ _id: 123, someData: 'abc' }, { _id: 456, someData: 'xyz' }]);
+
+      const endpoint = this.runner().endpoint('myEndpoint');
+      endpoint.execute(null, (err, result) => {
+        should.not.exist(err);
+        result.should.containDeep([{ _id: 123, someData: 'abc' }, { _id: 456, someData: 'xyz' }]);
+        return done();
+      });
+    });
+
+    it('should include empty object as body if no body passed', (done) => {
+      nock('https://baas.kinvey.com')
+        .matchHeader('content-type', 'application/json')
+        .matchHeader('x-kinvey-api-version', '3')
+        .post(`/rpc/${environmentId}/custom/myEndpoint`, {})
+        .basicAuth({
+          user: environmentId,
+          pass: mastersecret
+        })
+        .reply(200, [{ _id: 123, someData: 'abc' }, { _id: 456, someData: 'xyz' }]);
+
+      const endpoint = this.runner().endpoint('myEndpoint');
+      endpoint.execute((err, result) => {
+        should.not.exist(err);
+        result.should.containDeep([{ _id: 123, someData: 'abc' }, { _id: 456, someData: 'xyz' }]);
+        return done();
+      });
+    });
+
+    it('should resolve endpoint if callback isn\'t passed', (done) => {
+      nock('https://baas.kinvey.com')
+        .matchHeader('content-type', 'application/json')
+        .matchHeader('x-kinvey-api-version', '3')
+        .post(`/rpc/${environmentId}/custom/myEndpoint`)
+        .basicAuth({
+          user: environmentId,
+          pass: mastersecret
+        })
+        .reply(200, [{ _id: 123, someData: 'abc' }, { _id: 456, someData: 'xyz' }]);
+
+      const endpoint = this.runner().endpoint('myEndpoint');
+      endpoint.execute()
+        .then((result) => {
+          result.should.containDeep([{ _id: 123, someData: 'abc' }, { _id: 456, someData: 'xyz' }]);
+          return done();
+        });
+    });
+
+    it('should execute using userContext', (done) => {
+      nock('https://baas.kinvey.com')
+        .matchHeader('content-type', 'application/json')
+        .matchHeader('x-kinvey-api-version', '3')
+        .matchHeader('authorization', authorization)
+        .post(`/rpc/${environmentId}/custom/myEndpoint`)
+        .reply(200, [{ _id: 123, someData: 'abc' }, { _id: 456, someData: 'xyz' }]);
+
+      const endpoint = this.runner({ useUserContext: true }).endpoint('myEndpoint');
+      endpoint.execute({}, (err, result) => {
+        should.not.exist(err);
+        result.should.containDeep([{ _id: 123, someData: 'abc' }, { _id: 456, someData: 'xyz' }]);
```

Comment:
Why not assign this test object to a variable (with dynamically-generated values) and use it both here and in the nock handler?